darts/utils/likelihood_models/classification.py

```diff
@@ -0,0 +1,159 @@
+from typing import Optional
+
+import numpy as np
+
+from darts.logging import get_logger, raise_log
+from darts.utils.likelihood_models.base import LikelihoodType
+from darts.utils.likelihood_models.sklearn import SKLearnLikelihood
+
+logger = get_logger(__name__)
+
+
+class ClassProbabilityLikelihood(SKLearnLikelihood):
+    def __init__(
+        self,
+        n_outputs: int,
+        random_state: Optional[int] = None,
+    ):
+        """
+        Classes probability likelihood.
+        This likelihood is used for classification tasks where the model predicts classes probability.
+
+        Parameters
+        ----------
+        n_outputs
+            The number of predicted outputs per model call. `1` if `multi_models=False`, otherwise
+            `output_chunk_length`.
+        random_state
+            Optionally, control the randomness of the sampling.
+        """
+        super().__init__(
+            likelihood_type=LikelihoodType.ClassProbability,
+            n_outputs=n_outputs,
+            random_state=random_state,
+            parameter_names=[],
+        )
+
+    def fit(self, model):
+        """
+        Fits the likelihood to the model.
+
+        Parameters
+        ----------
+        model
+            The model to fit the likelihood to. The model is expected to be fitted and have a `classes_` attribute.
+        """
+        if not hasattr(model, "classes_"):
+            raise_log(
+                ValueError(
+                    "The model must have a `classes_` attribute to fit the likelihood."
+                ),
+                logger,
+            )
+
+        classes = model.classes_
+        if type(classes) is not list:
```

Comment:
Are we sure that all models only return lists and not any other sequence type (like numpy array or others)?

```suggestion
        if not isinstance(classes, list):
```

Reply:
Actually all models following the sklearn interface should return an array. Only our MultiOutputClassifier returns a list of array. Since unifying our MultiOutputClassifier is not possible I treat all models as if they are MultiOutputClassifier with a single estimator. I'm not enforcing that the provided model should return an array on .classes_